store/sqlstore/supplier_reactions.go

```diff
@@ -206,8 +206,25 @@ const (
 			Id = :PostId`
 )
 
-func updatePostForReactions(transaction *gorp.Transaction, postId string) error {
-	_, err := transaction.Exec(UPDATE_POST_HAS_REACTIONS_QUERY, map[string]interface{}{"PostId": postId, "UpdateAt": model.GetMillis()})
+func updatePostForReactionsOnDelete(transaction *gorp.Transaction, postId string) error {
+	_, err := transaction.Exec(UPDATE_POST_HAS_REACTIONS_ON_DELETE_QUERY, map[string]interface{}{"PostId": postId, "UpdateAt": model.GetMillis()})
+
+	return err
+}
+
+func updatePostForReactionsOnInsert(transaction *gorp.Transaction, postId string) error {
+	_, err := transaction.Exec(
+		`UPDATE
+			Posts
+		SET
+			UpdateAt = (CASE
```

Comment:
We are generally DB bound, so can we do this logic on the server stead of the DB?

Reply:
Yes, we can, because this query is based on the UPDATE_POST_HAS_REACTIONS_ON_DELETE_QUERY preexisting query, which is executed in MySQL and PostgreSQL. And doing it on server side, we can be sure of some degree of atomicity.

Reply:
Here I understood that you were asking if we can use the CASE construction in the DB server side. And we can, and actually is already done in the server side.

Reply:
Actually I would rather them on the API server side. Generally we run into more performance issues with the DB.

Reply:
Finally, I realize that the query can be simpler. No server or DB complexity needed, just a plain update with a where.